### PR TITLE
Convert :yank arguments to variables

### PR DESCRIPTION
**Pull request by blyxxyz**
Two of `:yank`'s possible arguments already existed as variables, and the other three were suitable for becoming variables. `:yank` now takes arbitrary text as an argument and the bindings pass the variables to it.

This adds `{title}`, `{domain}` and `{selection}`.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Oh, yet another big change on how `:yank` works :laughing: 

@mlochbaum want to help with reviewing it by any chance?

File: doc/help/commands.asciidoc [hunk 1]
```diff
@@ -894,13 +897,13 @@ Syntax: +:yank [*--sel*] [*--keep*] ['what']+
...
-* +'what'+: What to yank. 
+* +'what'+: What to yank. Typically one of these variables: 
 
- - `url`: The current URL.
- - `pretty-url`: The URL in pretty decoded form.
- - `title`: The current page's title.
- - `domain`: The current scheme, domain, and port number.
- - `selection`: The selection under the cursor.
+ - `{url}`: The current URL.
+ - `{pretty-url}`: The URL in pretty decoded form.
```
Comment (lahwaacz) [inline doc/help/commands.asciidoc]: This should be `{url:pretty}` :wink:

**Comment by The-Compiler** [comment]
How do we handle escaping for variables? That seems like the elephant in the room :wink:

@blyxxyz what's your plan regarding #1861? If we don't get that fixed in the next few days/weeks, I'd rather hold off more variable changes and probably revert the existing ones so they're out of v0.9.0...

File: qutebrowser/commands/runners.py [hunk 2]
```diff
@@ -57,6 +84,10 @@ def replace_variables(win_id, arglist):
             QUrl.RemovePassword),
         'clipboard': utils.get_clipboard,
         'primary': lambda: utils.get_clipboard(selection=True),
+        'title': lambda: tabbed_browser.page_title(
+            tabbed_browser.currentIndex()),
+        'domain': lambda: _current_domain(tabbed_browser),
```
Comment (code-hunger) [inline qutebrowser/commands/runners.py]: The whole `_current_domain` function can be stripped down to `url.toString(QUrl::RemovePath)`
(not sure about the python equivalent, see http://doc.qt.io/qt-5/qurl.html#UrlFormattingOption-enum

Comment (The-Compiler) [inline qutebrowser/commands/runners.py]: More like `QUrl.RemovePath | QUrl.RemoveRemoveAuthority | QUrl.RemoveQuery | QUrl.RemoveFragment` I'd say?

**Comment by The-Compiler** [comment]
I'll look at this and #1937 (which will conflict, I think?) as soon as there's a way to escape the commandline variables, see #1861 and #2017.

However, it'll probably take me some time until I get to that, so I'm closing this for now. Sorry, and thanks for your contribution anyways! I added the patch to #901 to ensure nothing gets lost.

**Pull request closed without merging**